exchanges/deribit/deribit_websocket_eps.go

```diff
@@ -0,0 +1,2174 @@
+	} else if startTime.After(endTime) {
+		return 0, errStartTimeCannotBeAfterEndTime
+	}
+	if endTime.IsZero() {
+		endTime = time.Now()
+	}
+	input := &struct {
+		Instrument     string `json:"instrument_name"`
+		StartTimestamp int64  `json:"start_timestamp"`
+		EndTimestamp   int64  `json:"end_timestamp"`
+	}{
+		Instrument:     instrument,
+		StartTimestamp: startTime.UnixMilli(),
+		EndTimestamp:   endTime.UnixMilli(),
+	}
+	var resp float64
+	return resp, d.SendWSRequest(getFundingRateValue, input, &resp, false)
+}
+
+// WSRetriveHistoricalVolatility retrives historical volatility data
+func (d *Deribit) WSRetriveHistoricalVolatility(currency string) ([]HistoricalVolatilityData, error) {
+	currency = strings.ToUpper(currency)
+	if currency != currencyBTC && currency != currencyETH && currency != currencySOL && currency != currencyUSDC {
+		return nil, errInvalidCurrency
+	}
+	input := &struct {
+		Currency string `json:"currency"`
+	}{
+		Currency: currency,
+	}
+	var data [][2]interface{}
+	err := d.SendWSRequest(getHistoricalVolatility, input, &data, false)
+	if err != nil {
+		return nil, err
+	}
+	resp := make([]HistoricalVolatilityData, len(data))
+	for x := range data {
+		timeData, ok := data[x][0].(float64)
+		if !ok {
+			return resp, fmt.Errorf("%v WSRetriveHistoricalVolatility: %w for time", d.Name, errTypeAssert)
+		}
+		val, ok := data[x][1].(float64)
+		if !ok {
+			return resp, fmt.Errorf("%v WSRetriveHistoricalVolatility: %w for val", d.Name, errTypeAssert)
+		}
+		resp[x] = HistoricalVolatilityData{
+			Timestamp: timeData,
+			Value:     val,
+		}
+	}
+	return resp, nil
+}
+
+// WSRetriveCurrencyIndexPrice the current index price for the instruments, for the selected currency through the websocket connection.
+func (d *Deribit) WSRetriveCurrencyIndexPrice(currency string) (map[string]float64, error) {
+	currency = strings.ToUpper(currency)
+	if currency != currencyBTC && currency != currencyETH && currency != currencySOL && currency != currencyUSDC {
+		return nil, errInvalidIndexPriceCurrency
+	}
+	input := &struct {
+		Currency string `json:"currency"`
+	}{
+		Currency: currency,
+	}
+	var resp map[string]float64
+	return resp, d.SendWSRequest(getCurrencyIndexPrice, input, &resp, false)
+}
+
+// WSRetriveIndexPrice retrives price data for the requested index through the websocket connection.
+func (d *Deribit) WSRetriveIndexPrice(index string) (*IndexPriceData, error) {
+	indexNames := map[string]bool{"ada_usd": true, "avax_usd": true, "btc_usd": true, "eth_usd": true, "dot_usd": true, "luna_usd": true, "matic_usd": true, "sol_usd": true, "usdc_usd": true, "xrp_usd": true, "ada_usdc": true, "avax_usdc": true, "btc_usdc": true, "eth_usdc": true, "dot_usdc": true, "luna_usdc": true, "matic_usdc": true, "sol_usdc": true, "xrp_usdc": true, "btcdvol_usdc": true, "ethdvol_usdc": true}
+	if !indexNames[index] {
+		return nil, errUnsupportedIndexName
+	}
```

Comment:
I think this check should be removed and have the exchange do the validation as this list of indexes can change